core/blockchain.go

```diff
@@ -1341,6 +1398,11 @@ func (bc *BlockChain) writeBlockWithState(block *types.Block, receipts []*types.
 	if err != nil {
 		return err
 	}
+	// If node is running in path mode, skip explicit gc operation
+	// which is unnecessary in this mode.
```

Comment:
Nice. Later on (not this PR), it might be neat to move the gc-stuff away. 
So we have 
```
if bc.triedb.Scheme() == rawdb.PathScheme {
    return nil
}
// Hashdb requires explicit gc / flushing
blockchain.doHashdbGc(a,b,c...)
```